app/controllers/telegram_bot_controller.rb

```diff
@@ -0,0 +1,45 @@
+class TelegramBotController < Telegram::Bot::UpdatesController
+  include Telegram::Bot::UpdatesController::MessageContext
+  def start!(*)
+    respond_with :message, text: 'Hi. Entry your /email example@mail.com and then /password example'
+  end
+
+  def email!(email)
+    respond_with :message, text: 'Success'
+    session['email'] = email
+  end
+
+  def password!(password)
+    session['password'] = password
+    if user_valid?
+      respond_with :message, text: 'Success. Entry /channels to see all channels'
+    else
+      respond_with :message, text: 'Not valid email or password'
+    end
+  end
+
+  def channels!
+    if user_valid?
+      respond_with :message, text: channels.pluck(:name).join("\n")
+    else
+      respond_with :message, text: 'Not valid email or password'
```

Comment:
that's should probably be a `401` saying `you have no access` or to confuse `400` `not found` (as github does) so that hacker wouldn't know either creds are incorrect or page not exists